test/extended/machines/cluster.go

```diff
@@ -100,4 +119,293 @@ var _ = g.Describe("[sig-node] Managed cluster", func() {
 		err := prometheus.RunQueries(context.TODO(), oc.NewPrometheusClient(context.TODO()), tests, oc)
 		o.Expect(err).NotTo(o.HaveOccurred())
 	})
+
+	g.It("should verify that nodes have exactly two boots [Late]", func() {
+		ctx := context.Background()
+
+		// This test is applicable for SNO installations only
+		configClient, err := configv1client.NewForConfig(oc.AdminConfig())
+		o.Expect(err).NotTo(o.HaveOccurred())
+		infrastructure, err := configClient.ConfigV1().Infrastructures().Get(ctx, "cluster", metav1.GetOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+		if infrastructure.Status.ControlPlaneTopology != configv1.SingleReplicaTopologyMode {
+			return
+		}
+
+		createTestBed(ctx, oc)
+		defer deleteTestBed(ctx, oc)
+
+		// List all nodes
+		nodes, err := oc.KubeClient().CoreV1().Nodes().List(ctx, metav1.ListOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(nodes.Items).NotTo(o.HaveLen(0))
+
+		expected := make(map[string]int)
+		actual := make(map[string]int)
+		errs := make([]error, 0)
+
+		// Find a number of rendered-* machineconfigs for masters and workers
+		// Each rendered config adds expected reboot
+		expectedReboots := map[string]int{
+			"worker": 0,
+			"master": 0,
+		}
+		dynamicClient, err := dynamic.NewForConfig(oc.KubeFramework().ClientConfig())
+		o.Expect(err).NotTo(o.HaveOccurred())
+		machineConfigGroupVersionResource := schema.GroupVersionResource{
+			Group: "machineconfiguration.openshift.io", Version: "v1", Resource: "machineconfigs",
+		}
+		mcList, err := dynamicClient.Resource(machineConfigGroupVersionResource).List(ctx, metav1.ListOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+		for i := range mcList.Items {
+			mcName := mcList.Items[i].GetName()
+			for prefix := range expectedReboots {
+				if strings.HasPrefix(mcName, fmt.Sprintf("rendered-%s-", prefix)) {
+					expectedReboots[prefix] += 1
+				}
+			}
+		}
+
+		// List nodes, set actual and expected number of reboots
+		for _, node := range nodes.Items {
+			// Examine only the nodes which are available at the start of the test
+			if !slices.Contains(staticNodeNames, node.Name) {
+				continue
+			}
+			expectedRebootsForNodeRole, ok := expectedReboots[getNodeRole(&node)]
+			o.Expect(ok).To(o.BeTrue())
+			expected[node.Name] = expectedRebootsForNodeRole
+			nodeReboots, err := getNumberOfBootsForNode(oc.KubeClient(), node.Name, 0)
+			if err != nil {
+				errs = append(errs, err)
+			}
+			actual[node.Name] = nodeReboots
+		}
+
+		o.Expect(errs).To(o.BeEmpty())
+		o.Expect(actual).To(o.Equal(expected))
+	})
 })
+
+// getNodeRole reads node labels and returns either "worker" or "master"
+func getNodeRole(node *corev1.Node) string {
```

Comment:
Do SNO nodes have multiple node-role labels?

Reply:
I don't think it possible to add more labels to masters, but you can have SNO + worker which acts as infra node